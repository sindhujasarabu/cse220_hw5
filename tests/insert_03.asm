.data
_num_students: .word 4
_id_list: .word 3126310, 1264321, 553612, 99999
_credits_list: .word 314, 511, 290, 99
_names: .ascii "Kevin T. McDonnell\0Wolfie Seawolf\0Kelly Chen\0New Student\0"

.align 2
_records: .space 32  # Must be 8 * num_students
_table: .space 12  # Must be 4 * table_size
_table_size: .word 3
_record_to_insert: .word 0  # Filled later

.text
main:
# Initialize student records array
lw $a0, _num_students
la $a1, _id_list
la $a2, _credits_list
la $a3, _names

addi $sp, $sp, -4
la $t0, _records
sw $t0, 0($sp)

jal init_student_array
addi $sp, $sp, 4

# Populate the table with existing records to nearly full
la $s1, _table
la $s0, _records

sw $s0, 0($s1)   # Store at index 0
addi $s0, $s0, 8
sw $s0, 4($s1)   # Store at index 1
addi $s0, $s0, 8
sw $s0, 8($s1)   # Store at index 2
addi $s0, $s0, 8

sw $s0, _record_to_insert

# Now include the tester which will call the insert function
.include "p4_tester.asm"
