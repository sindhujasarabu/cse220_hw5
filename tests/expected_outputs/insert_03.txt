$v0: -1
3126310 314 Kevin T. McDonnell
1264321 511 Wolfie Seawolf
553612 290 Kelly Chen
